zerver/worker/queue_processors.py

```diff
@@ -246,11 +262,17 @@ def do_consume(self, consume_func: Callable[[List[Dict[str, Any]]], None],
                 self.update_statistics(self.get_remaining_queue_size())
 
             time_start = time.time()
-            consume_func(events)
+            if self.MAX_CONSUME_SECONDS and self.ENABLE_TIMEOUTS:
+                signal.signal(signal.SIGALRM, functools.partial(timer_expired, self.MAX_CONSUME_SECONDS, len(events)))
+                signal.alarm(self.MAX_CONSUME_SECONDS * len(events))
+                consume_func(events)
+                signal.alarm(0)
```

Comment:
This needs to be in a `finally` block to prevent leaking an alarm if `consume_func` raises an exception for some other reason. #16486.

Reply:
Thanks for this fix!